File: js/task2b/S2_FOR_02.js
// S2_FOR_02 - Find first match

function findFirstEven(nums) {
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] % 2 === 0) {
      return nums[i];
    }
  }
  return null;
}

// tests
console.log(findFirstEven([1, 3, 5, 8, 9]));  // 8
console.log(findFirstEven([2, 4, 6]));         // 2
console.log(findFirstEven([1, 3, 5, 7]));      // null
console.log(findFirstEven([]));                 // null

File: python/task2b/S2_FOR_02.py
# S2_FOR_02 - Find first even (no next() with generator)

def find_first_even(nums):
    for n in nums:
        if n % 2 == 0:
            return n
    return None

# Tests
print(f"find_first_even([1, 3, 4, 5]): {find_first_even([1, 3, 4, 5])}")
print(f"find_first_even([1, 3, 5, 7]): {find_first_even([1, 3, 5, 7])}")
print(f"find_first_even([2, 4, 6]): {find_first_even([2, 4, 6])}")
print(f"find_first_even([]): {find_first_even([])}")
